### ISSUE TITLE
Add support for RDS Storage auto scaling

### ISSUE THREAD

**Issue by edhgoose** [terraform-providers/terraform-provider-aws#9076]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

AWS have announced auto scaling of storage for some RDS instance types. 

https://aws.amazon.com/about-aws/whats-new/2019/06/rds-storage-auto-scaling/

### New or Affected Resource(s)

* aws_rds

### References

https://docs.aws.amazon.com/AmazonRDS/latest/UserGuide/USER_PIOPS.StorageTypes.html#USER_PIOPS.Autoscaling

**Comment by ewbankkit** [comment]
Requires:
* https://github.com/terraform-providers/terraform-provider-aws/pull/9072

**Comment by mwarkentin** [comment]
I'm curious how this will look like on plans if the AWS storage has diverged from your initial defined storage for the RDS instance?

**Comment by bflad** [comment]
@mwarkentin presumably Terraform will show a difference for the argument unless [`ignore_changes`](https://www.terraform.io/docs/configuration/resources.html#ignore_changes) is used, e.g.

```hcl
resource "aws_db_instance" "example" {
  allocated_storage = 123

  # ... other configuration ...

  # Prevent RDS storage auto scaling from recreating the resource
  lifecycle {
    ignore_changes = ["allocated_storage"]
  }
}
```

When this feature is implemented, we will likely want to add this to the resource documentation.

**Comment by bflad** [comment]
It looks like this will be a new `max_allocated_storage` argument, where storage autoscaling is enabled if it is configured:

```hcl
resource "aws_db_instance" "example" {
  allocated_storage     = 100
  max_allocated_storage = 200

  # ... other configuration ...
}
```

Given this implementation, we might be able to ignore the `allocated_storage` argument differences automatically within the resource if `max_allocated_storage` is defined.

**Comment by mwarkentin** [comment]
Ah nice, I think that makes sense.

**Comment by bflad** [comment]
Pull request submitted: #9087

**Comment by bflad** [comment]
Support for the new `max_allocated_storage` argument has been merged and will release with version 2.17.0 of the Terraform AWS Provider, likely later today. 👍

**Comment by bflad** [comment]
This has been released in [version 2.17.0 of the Terraform AWS provider](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md#2170-june-28-2019). Please see the [Terraform documentation on provider versioning](https://www.terraform.io/docs/configuration/providers.html#provider-versions) or reach out if you need any assistance upgrading.

For further feature requests or bug reports with this functionality, please create a [new GitHub issue](https://github.com/terraform-providers/terraform-provider-aws/issues/new/choose) following the template for triage. Thanks!
